### PR TITLE
Wait for credential offer shortly following a new connection

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

When a new connection is formed, this new workflow will wait for a short period of time (a slightly exaggerated 14sec for testing) and if an offer is made automatically display the offer.

# Related Issues

n/a

# Pull Request Checklist

This is just a reminder about the most common mistakes. Please make sure that you tick all _appropriate_ boxes. But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this).
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components.
- [x] Run prettier: `npm run style-format`
- [x] Updated **documentation** for changed code and new or modified features.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by jleach** [comment]
@amanji I moved the logic for accepting a credential into the "AcceptOffer" screen. I prefer it there as that screen is it separates concerns. The offer screen just shows the credential being offered while the accept offer screen deals with that part of the workflow. I didn't do the same for decline for comparison. If you think it's a good approach I can do that as part of a separate PR as it's more about technical debt rather than functionality.

**Comment by jleach** [comment]
I rebased and squashed much of this PRs history because I was having trouble rebasing in previous changes to `main`. The history is still preserved as per above. Tested after rebasing and it seems fine. Just a few tests needed updating.

File: App/screens/ConnectionModal.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  delayMessageContainer: {
+    marginTop: 30,
+  },
+  delayMessageText: {
+    textAlign: 'center',
+    marginBottom: 30,
+  },
+})
+
+type ConnectionModalProps = StackScreenProps<DeliveryStackParams, Screens.ConnectionModal>
```
Comment (amanji) [inline App/screens/ConnectionModal.tsx]: Why do we need both the modal and the screen?

File: App/screens/Scan.tsx [hunk 2]
```diff
@@ -45,34 +29,50 @@ const Scan: React.FC<ScanProps> = ({ navigation }) => {
...
-        type: DispatchAction.ConnectionPending,
-        payload: [{ ConnectionPending: false }],
+  const handleInvitation = async (url: string): Promise<void> => {
+    try {
+      const connectionRecord = await agent?.connections.receiveInvitationFromUrl(url, {
+        autoAcceptConnection: true,
       })
 
+      if (!connectionRecord?.id) {
+        throw new Error(t('Scan.ConnectionNotFound'))
```
Comment (amanji) [inline App/screens/Scan.tsx]: Should we throw a BifoldError here or generic?

**Review by amanji — DISMISSED** [review]

File: App/screens/ConnectionModal.tsx (hunk 1 above)
Comment (jleach) [reply App/screens/ConnectionModal.tsx]: Will cleanup. I think its left over from one of my previous incarnations of this feature.

File: App/screens/Scan.tsx (hunk 2 above)
Comment (jleach) [reply App/screens/Scan.tsx]: Will cleanup.

**Review by amanji — DISMISSED** [review]

**Review by amanji — DISMISSED** [review]

**Review by amanji — APPROVED** [review]

**Pull request merged**